Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -0,0 +1,62 @@
+{ Expression } = require './expression'
+{ FunctionRef } = require './reusable'
+{ ValueSet, Code } = require '../datatypes/datatypes'
+{ build } = require './builder'
+
+module.exports.Quantity = class Quantity extends Expression
+  constructor: (json) ->
+    super
+    @unit = json.unit
+    @value = json.value
+
+  exec: (ctx) ->
+    @  
+  sameOrBefore: (other) ->
+    if other instanceof Quantity and other.unit == @unit then @value <= other.value else null
+
+  sameOrAfter: (other) ->
+    if other instanceof Quantity and other.unit == @unit then @value >= other.value else null
+
+  after: (other) ->
+    if other instanceof Quantity and other.unit == @unit then @value > other.value else null
+
+  before: (other) ->
+    if other instanceof Quantity and other.unit == @unit then @value < other.value else null
+
+time_units = {'years':'year',  'months': 'month',  'days' :'day', 'minutes': 'minute', 'seconds':'seconds', 'milliseconds' : 'millisecond' }
+
+clean_unit = (units) ->
+  if time_units[units] then time_units[units] else units
+
+module.exports.createQuantity = (value,unit) ->
+  new Quantity({value: value, unit: unit})
+
+module.exports.doAddition = (a,b) ->
+  if a instanceof Quantity and b instanceof Quantity
+    if a.unit == b.unit 
+      new Quantity({unit: a.unit, value: a.value + b.value})
+  else
```

Comment:
Ah.  Changing my mind here too.  I see that the way it is called, it can only be called for Quantity or DateTime -- and given that it is in a Quantity class, it makes sense that it only works in some contexts.  Still, there may be some potential for other cleanup that would make this a bit nicer (like a Quantity class that supports add, etc).  I'll wait until the ucum stuff is in here to consider any refactoring like that.